lib/streams/compound-duplex.js

```diff
@@ -63,12 +63,40 @@ function CompoundDuplex(writable, readable, options) {
 		self.push(null);
 	});
 
-	this.on('finish', function() {
-		writable.end();
+	readable.on('finish', function() {
```

Comment:
Why is this needed, and how is it a substitute for the original version?

The function of the original version was to propagate the EOF from the duplex stream (as the base class) to the next link in the pipeline (ie, the readable end of the compound duplex).

The changed version doesn't seem to do that at all; I can't tell what it's doing.  Why is it listening for 'finish' on a readable stream, when Readable doesn't even emit 'finish' events?